src/engine/renderer/tr_shade.cpp

```diff
@@ -2297,6 +2312,8 @@ static void Render_liquid( int stage )
 
 	GLimp_LogComment( "--- Render_liquid ---\n" );
 
+	bool specularMapping = r_specularMapping->integer; // TODO: specular map?
```

Comment:
You did it, right?

Reply:
yes I taken care of this question, I will remove that comment
note that liquid shader does not use specular map from file, the specular one is somewhat computed. That prevents factorisation a lot, especially we have to define all the specular map in the glsl files that relies on `reliefMapping_fp.glsl` instead of declaring it in `reliefMapping_fp.glsl` and doing some compute there…).

Reply:
(hence the bizarre macro with hardcoded names in `reliefMapping_fp.glsl` to avoid a pile of ifdef in all other files)…